jakarta-eclipse/org.eclipse.lsp4jakarta.core/src/org/jakarta/jdt/cdi/ManagedBeanDiagnosticsCollector.java

```diff
@@ -102,54 +102,104 @@ public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnosti
+                         * https://jakarta.ee/specifications/cdi/3.0/jakarta-cdi-spec-3.0.html#
+                         * declaring_producer_method
+                         * https://jakarta.ee/specifications/cdi/3.0/jakarta-cdi-spec-3.0.html#
+                         * declaring_injected_field
+                         * https://jakarta.ee/specifications/cdi/3.0/jakarta-cdi-spec-3.0.html#
+                         * declaring_initializer
+                         */
+                        
+                        // A single field cannot have the same
+                        diagnostics.add(createDiagnostic(field, unit,
                                 "@Produces and @Inject annotations cannot be used on the same field or property",
                                 ManagedBeanConstants.DIAGNOSTIC_CODE_PRODUCES_INJECT));
                     }
+
                 }
 
-                for (IField field : type.getFields()) {
-                    IAnnotation ProducesAnnotation = null;
-                    IAnnotation InjectAnnotation = null;
-                    for (IAnnotation annotation : field.getAnnotations()) {
-                        if (annotation.getElementName().equals(ManagedBeanConstants.PRODUCES))
-                            ProducesAnnotation = annotation;
-                        if (annotation.getElementName().equals(ManagedBeanConstants.INJECT))
-                            InjectAnnotation = annotation;
+                for (IMethod method : type.getMethods()) {
+                    /**
+                     * https://jakarta.ee/specifications/cdi/3.0/jakarta-cdi-spec-3.0.html#declaring_bean_scope
+                     * A bean class or producer method or field may specify at most one scope type
+                     * annotation. If a bean class or producer method or field specifies multiple
+                     * scope type annotations, the container automatically detects the problem and
+                     * treats it as a definition error.
+                     * 
+                     * Here we only look at the methods.
+                     */
+                    List<IAnnotation> methodAnnotations = Arrays.asList(method.getAnnotations());
+                    List<String> methodScopes = getScopeAnnotations(method);
+
+                    boolean isProducerMethod = methodAnnotations.stream()
+                            .anyMatch(annotation -> annotation.getElementName().equals(ManagedBeanConstants.PRODUCES));
+
+                    boolean isInjectMethod = methodAnnotations.stream()
+                            .anyMatch(annotation -> annotation.getElementName().equals(ManagedBeanConstants.INJECT));
+
+                    if (isProducerMethod && methodScopes.size() > 1) {
+                        diagnostics.add(createDiagnostic(method, unit,
+                                "A bean class or producer method or field may specify at most one scope type annotation.",
```

Comment:
Tailor the diagnostic message to clarify if it is a field/producer method/ bean class